website/markdown/posts/2020-05-29-version-1.9.0-speedy-gonzales/post.mdx

```diff
@@ -0,0 +1,105 @@
+---
+layout: post
+title: Tuist 1.9.0 - Speedy Gonzales
+categories: [Tuist, release, swift, project generation, xcode, apple, '1.9.0']
+excerpt: In this blog post we announce the latest version of Tuist, 1.9.0, which introduces significant improvements in the performance of the project generation.
+author: pepibumur
+---
+
+Hi there 👋,
+
+It has been quiet for the past few weeks,
+but we are back to announce a new release that we just baked,
+Tuist 1.9.0.
+As you might have noticed,
+from this release,
+we started giving them names.
+We not only do it for fun,
+but we think it is easier to remember a release.
+
+## Faster project generation
+
+1.9.0's name is Speedy Gonzales,
+which relates well with one of the most significant improvements with are shipping with this release:
+**faster project generation.**
+
+One of the features that make Tuist's project generation unique at scale is using Swift as a format for the manifest files.
+That allows developers to edit their projects using Xcode,
+where they can get auto-completion, documentation, and validations.
+Moreover,
+they can extract pieces of their manifest into a separate framework,
+`ProjectDescriptionHelpers`,
+which they can import and use from their manifest files.
+That's great,
+but it comes with costs.
+One of those is having to compile the files,
+which takes significantly longer than reading a YAML or JSON file from disk.
+[Kas](https://github.com/kwridan) noticed that and ventured into improving that.
+
+The result of this work is a project generation that is **approximately 98% faster**.
```

Comment:
its only the manifest loading process that is ~98% faster - not the entire project generation sadly :/ 

For a rough idea on generation speeds improvement it's worth looking at the benchmark results:

![Screen Shot 2020-05-29 at 11 17 36 AM](https://user-images.githubusercontent.com/11914919/83249122-06403f80-a19e-11ea-8c85-ec95adb8327f.png)